### PR TITLE
Update to React 18

### PR DESCRIPTION
**Pull request by fleonard**
## What does this do?

Update S'mores to React 18 and tweak peer deps to make it available to older versions.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,9 +1,6 @@
 module.exports = {
   stories: ['../src/**/*.stories.tsx'],
-  addons: [
-    '@storybook/addon-essentials',
-    'storybook-addon-performance/register',
```
Comment (fleonard) [inline .storybook/main.js:5]: Storybook performance doesn't yet support React 18 so I've removed it for now. We can add it back once they do the update

File: package.json [hunk 2]
```diff
@@ -39,12 +39,11 @@
     "@storybook/react": "^6.2.6",
     "@storybook/storybook-deployer": "^2.8.7",
     "@testing-library/jest-dom": "^5.11.4",
-    "@testing-library/react": "^12.0.0",
-    "@testing-library/react-hooks": "^8.0.0",
```
Comment (fleonard) [inline package.json:43]: `@testing-library/react-hooks` is now deprecated since `@testing-library/react` ship it's own version of `renderHook`

**Review by mickaelzhang — COMMENTED** [review]
Some component's doesn't needs `PropsWithChildren` like the `SupportMessage` or the `SearchInput` I think?

**Comment by fleonard** [comment]
@mickaelzhang yeah you are right. I've actually used a script to fix the PropsWithChildren so it might have added it to components that might not need it. I'll double check

**Review by mickaelzhang — DISMISSED** [review]

**Comment by fleonard** [comment]
@mickaelzhang I've removed `PropsWithChildren` since we were already exporting `children` already and you were actually right that using it was complaining in the IDE

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -15,4 +15,4 @@
 # This should make it easy to add new rules without breaking existing ones.
 
 # Global - These owners will be the default owners for everything in the repo.
-*           @ellajay @fleonard @habin-isa @jmuskett @josianed @mickaelzhang
+*           @ellajay @fleonard @habin-isa @josianed @mickaelzhang
```
Comment (mickaelzhang) [inline .github/CODEOWNERS:18]: Wondering if maybe we could change this to the frontend group?

File: package-lock.json [hunk 4]
```diff
@@ -127,9 +127,9 @@
       }
     },
     "node_modules/@babel/generator": {
-      "version": "7.20.2",
-      "resolved": "https://registry.npmjs.org/@babel/generator/-/generator-7.20.2.tgz",
-      "integrity": "sha512-SD75PMIK6i9H8G/tfGvB4KKl4Nw6Ssos9nGgYwxbgyTP0iX/Z55DveoH86rmUB/YHTQQ+ZC0F7xxaY8l2OF44Q==",
+      "version": "7.20.4",
+      "resolved": "https://registry.npmjs.org/@babel/generator/-/generator-7.20.4.tgz",
+      "integrity": "sha512-luCf7yk/cm7yab6CAW1aiFnmEfBJplb/JojV56MYEK7ziWfGmFlTfmL9Ehwfy4gFhbjBfWO1wj7/TuSbVNEEtA==",
```
Comment (mickaelzhang) [inline package-lock.json:132]: Is it normal that the package-lock.json has changed on this PR without an update on `package.json`?

Comment (fleonard) [reply package-lock.json:132]: no it's not I think the rebase messed everything up 😢 
I'll try to fix it

File: .github/CODEOWNERS (hunk 3 above)
Comment (fleonard) [reply .github/CODEOWNERS:18]: the problem is that in the FE team there are a lot of BE and I want to avoid spamming them 😁

**Review by mickaelzhang — APPROVED** [review]

**Pull request merged**